### PR TITLE
ENT-11440: Added nghttp2 as a vendored dependency we build as part of the hub package for apache http2 support

### PR DESCRIPTION
**Pull request by craigcomstock**
Ticket: ENT-11440
Changelog: title

together:
https://github.com/cfengine/masterfiles/pull/2944
https://github.com/cfengine/buildscripts/pull/1489
https://github.com/cfengine/core/pull/5610

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thank you for submitting a PR! Maybe @larsewi can review this?

**Comment by craigcomstock** [comment]
little typo in spec file, version -> nghttp2_version :p [Build Status](https://ci.cfengine.com/job/pr-pipeline/11108/)

**Comment by craigcomstock** [comment]
Good deal, failure gives me the information I need to exclude files in the RPM spec file:

```
00:00:58 Checking for unpackaged file(s): /usr/lib/rpm/check-files /home/jenkins/workspace/testing-pr/label/PACKAGES_HUB_x86_64_linux_redhat_7/nghttp2/BUILDROOT/cfbuild-nghttp2-0+215d80d+untested-1.x86_64
00:00:58 error: Installed (but unpackaged) file(s) found:
00:00:58    /var/cfengine/lib/libnghttp2.a
00:00:58    /var/cfengine/lib/libnghttp2.la
00:00:58    /var/cfengine/share/doc/nghttp2/README.rst
00:00:58    /var/cfengine/share/man/man1/h2load.1
00:00:58    /var/cfengine/share/man/man1/nghttp.1
00:00:58    /var/cfengine/share/man/man1/nghttpd.1
00:00:58    /var/cfengine/share/man/man1/nghttpx.1
00:00:58    /var/cfengine/share/nghttp2/fetch-ocsp-response
00:00:58 
00:00:58 
00:00:58 RPM build errors:
00:00:58     Installed (but unpackaged) file(s) found:
00:00:58    /var/cfengine/lib/libnghttp2.a
00:00:58    /var/cfengine/lib/libnghttp2.la
00:00:58    /var/cfengine/share/doc/nghttp2/README.rst
00:00:58    /var/cfengine/share/man/man1/h2load.1
00:00:58    /var/cfengine/share/man/man1/nghttp.1
00:00:58    /var/cfengine/share/man/man1/nghttpd.1
00:00:58    /var/cfengine/share/man/man1/nghttpx.1
00:00:58    /var/cfengine/share/nghttp2/fetch-ocsp-response
00:00:58 + status=failure
```

**Comment by craigcomstock** [comment]
try a build for rhel7hub and ubu20hub [Build Status](https://ci.cfengine.com/job/pr-pipeline/11109/)

**Comment by craigcomstock** [comment]
fixed rpm maybe [Build Status](https://ci.cfengine.com/job/pr-pipeline/11110/) (rhel7hub only)

**Comment by craigcomstock** [comment]
ok, I guess --enable-fpm also enables ZTS (zend thread safety) so the name of the extensions folder has changed from no-debug-non-zts-<date> to no-debug-zts-<date>

**Comment by craigcomstock** [comment]
another try, needed to add some new files in php package for fpm [Build Status](https://ci.cfengine.com/job/pr-pipeline/11113/)

**Comment by craigcomstock** [comment]
Try with fpm configuration setup properly. One thing left: php-fpm as a service

[Build Status](https://ci.cfengine.com/job/pr-pipeline/11115/)

**Comment by craigcomstock** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/11116/)

**Comment by craigcomstock** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/11120/)

**Comment by craigcomstock** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/11121/)

File: deps-packaging/php/www.conf [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+
+; The desired minimum number of idle server processes.
+; Note: Used only when pm is set to 'dynamic'
+; Note: Mandatory when pm is set to 'dynamic'
+pm.min_spare_servers = 1
+
+; The desired maximum number of idle server processes.
+; Note: Used only when pm is set to 'dynamic'
+; Note: Mandatory when pm is set to 'dynamic'
+pm.max_spare_servers = 3
```
Comment (aleksandrychev) [inline deps-packaging/php/www.conf:142]: I'd use these as min values

```suggestion
pm.max_children = 80

; The number of child processes created on startup.
; Note: Used only when pm is set to 'dynamic'
; Default Value: (min_spare_servers + max_spare_servers) / 2
pm.start_servers = 20

; The desired minimum number of idle server processes.
; Note: Used only when pm is set to 'dynamic'
; Note: Mandatory when pm is set to 'dynamic'
pm.min_spare_servers = 10

; The desired maximum number of idle server processes.
; Note: Used only when pm is set to 'dynamic'
; Note: Mandatory when pm is set to 'dynamic'
pm.max_spare_servers = 30
```

**Comment by aleksandrychev** [comment]
with master/mp/nova [Build Status](https://ci.cfengine.com/job/pr-pipeline/11136/)

**Comment by craigcomstock** [comment]
get a rhel7hub package to see where we are at: [Build Status](https://ci.cfengine.com/job/pr-pipeline/11185/)

**Comment by craigcomstock** [comment]
moving on to another PR...

**Pull request closed without merging**